scripts/sdk-api-diff.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+set -e
+
+usage="--------------------------------------------
+Description: Compares API differences between two module versions.
+
+Usage: sdk-api-diff <module name> <module A path> <module B path> [OPTIONS]
+
+Options:
+  --target: Target platform and version e.g. x86_64-apple-ios14.0-simulator or arm64-apple-ios14.0
+    (Default: x86_64-apple-ios14.0-simulator)
+  --sdkpath: Path to target's SDK.
+    (Default: output of 'xcrun --sdk iphonesimulator --show-sdk-path')"
+
+# default values
+TARGET_NAME=x86_64-apple-ios14.0-simulator
+SDK_DIR=`xcrun --sdk iphonesimulator --show-sdk-path`
+
+for i in "$@"; do
+  case $i in
+    --target=*)
+      TARGET_NAME="${i#*=}"
+      shift
+      ;;
+    --sdkpath=*)
+      SDK_DIR="${i#*=}"
+      shift
+      ;;
+    --help)
+      printf -- "$usage"
+      exit 0
+      ;;
+    *)
+      # unknown option
+      ;;
+  esac
+done
+
+if [ -z ${1+x} ] || [ -z ${2+x} ] || [ -z ${3+x} ]; then
+    echo "Error: Missing required arguments"
+    printf -- "$usage"
+    exit 1
+fi
+if [ ! -d $2 ]; then
+    echo "Invalid path: $2"
+    exit 1
+fi
+if [ ! -d $3 ]; then
+    echo "Invalid path: $3"
+    exit 1
+fi
+REQUEST_TEMPALTE="sdk-api-diff-request-tempalte.yml"
+if [ ! -f $REQUEST_TEMPALTE ]; then
+    echo "Missing $REQUEST_TEMPALTE file"
+    exit 1
+fi
+
+hash jq 2> /dev/null || { echo >&2 "ERROR: jq must be installed"; exit 1; }
+hash sourcekitten 2> /dev/null || { echo >&2 "ERROR: sourcekitten must be installed"; exit 1; }
+
+MODULE_NAME=$1
+MODULE_A_DIR=$2
+MODULE_B_DIR=$3
+
+# Export vars for yaml template
+export TARGET_NAME
+export SDK_DIR
+export MODULE_NAME
+export MODULE_DIR=$MODULE_A_DIR
+
+# Replace the {{KEY}} in yaml file with the values from the environment variables
+cat $REQUEST_TEMPALTE | perl -pe 's/\{\{(\w+)\}\}/$ENV{$1}/eg' > requestA.yml
+
+export MODULE_DIR=$MODULE_B_DIR
+cat $REQUEST_TEMPALTE | perl -pe 's/\{\{(\w+)\}\}/$ENV{$1}/eg' > requestB.yml
```

Comment:
TEMPALTE -> TEMPLATE

Reply:
Maybe add an explanatory comment for the regex

Reply:
There's a comment for identical regex that's above. Should it be more detailed?